File: backend/python-flask/api/rabbit_producer.py
# Why this file exists:
# - Minimal RabbitMQ producer (publish to task.created)

import pika

def publish_task_created(title: str):
    conn = pika.BlockingConnection(pika.ConnectionParameters("localhost"))
    ch = conn.channel()

    ch.queue_declare(queue="task.created")
    ch.basic_publish(exchange="", routing_key="task.created", body=title)

    conn.close()

File: backend/nestjs/api/rabbit.producer.ts
// Why this file exists:
// - Minimal RabbitMQ producer (publish to task.created)

import amqp from "amqplib";

export async function publishTaskCreated(title: string) {
  const conn = await amqp.connect("amqp://localhost");
  const ch = await conn.createChannel();

  await ch.assertQueue("task.created");
  ch.sendToQueue("task.created", Buffer.from(title));

  await ch.close();
  await conn.close();
}
